files/image_config/monit/container_checker

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+"""
+container_checker
+
+This script is intended to be run by Monit. It will write an alerting message into
+syslog if it found containers which were expected to run but were not running. At
+the same time, if some containers were unexpected to run, it also writes an alerting
+syslog message. Note that if print(...) statement in this script was executed, the
+string in it will be appended to Monit syslog messages.
+
+The following is an example in Monit configuration file to show how Monit will run
+this script:
+
+check program container_checker with path "/usr/bin/container_checker"
+    if status != 0 for 5 times within 5 cycles then alert repeat every 1 cycles
+"""
+
+import subprocess
+import sys
+
+import swsssdk
+from sonic_py_common import multi_asic
+
+
+def get_command_result(command):
+    """
+    @summary: This function will execute the command and return the resulting output.
+    @return: A string which contains the output of command.
+    """
+    command_stdout = ""
+
+    try:
+        proc_instance = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                         shell=True, universal_newlines=True)
+        command_stdout, command_stderr = proc_instance.communicate()
+        if proc_instance.returncode != 0:
+            print("Failed to execute the command '{}'. Return code: '{}'".format(
+                command, proc_instance.returncode))
+            sys.exit(1)
+    except (OSError, ValueError) as err:
+        print("Failed to execute the command '{}'. Error: '{}'".format(command, err))
+        sys.exit(2)
+
+    return command_stdout.split("\n")[:-1]
```

Comment:
Why remove the last line here (`[:-1]`)?

Reply:
I think it makes more sense to `.rstrip()` before splitting rather than blindly delete the last line. This is written as a general-purpose function. Not all command outputs may end with a blank line, in which case you would lose the last line of output.

Reply:
Great suggestion!

Reply:
My point was that not all command output is guaranteed to end in a `\n`; in those cases `command_stdout.split("\n")[:-1]` would lose the last line.

Reply:
Ohh, Got it. Thanks!